CHANGES.rst

```diff
@@ -1,6 +1,218 @@
+  cases where the ``"*"`` appears in a terminal position. For example,
+  ``observe("foo:*")`` is supported, but ``observe("*:foo")`` is not.
+  (#1525, #1496)
+* The ``Any`` trait type now supports a ``factory`` argument (with accompanying
+  ``args`` and ``kw`` arguments). This can be used to specify a per-instance
+  default, for example with ``Any(factory=dict)``. (#1557, #1558)
+* Add a new member ``DefaultValue.disallow`` to the ``DefaultValue``
+  enumeration, to be used for trait types that don't have a meaningful default.
+* Warn on a bad signature for an ``observe``-decorated method, to catch the
+  common error of not providing the ``event`` parameter. (#1529)
+* Support the ``"qt"`` toolkit name as a synonym for ``"qt4"``. (#1436)
+* The ``Date``, ``Datetime`` and ``Time`` trait types have a new argument
+  ``allow_none``. In the future, these trait types will not accept ``None``
+  unless ``allow_none=True`` is specified. (#1432)
+* The ``Date`` trait type has a new argument ``allow_datetime``. In the future,
+  ``datetime`` instances will not be valid for a ``Date`` trait unless
+  ``allow_datetime`` is specified. (#1429)
+
+
+Performance
+~~~~~~~~~~~
+
+* Previously compiled ``ObserverExpression``s and observe mini-language strings
+  are now cached, to speed up compilation of commonly-used expressions.
+  (#1528, #1516)
+* Simplify equality definition on ``ObserverExpression``. (#1517)
+* Add ``__slots__`` to ``ObserverExpression``, ``ObserverGraph`` and related
+  classes. (#1513, #1515)
+* The ``on_trait_change`` method has been sped up by almost a factor of two,
+  by removing unnecessary internal usage of Traits in the parsing and listener
+  functionality. (#1493, #1492, #1491, #1490)
+
+
+Changes
+~~~~~~~
+
+* An invalid static default value in a ``PrefixList`` or ``PrefixMap`` trait
+  declaration now raises ``ValueError`` rather than ``TraitError``. (#1564)
+* ``PrefixList`` and ``PrefixMap`` no longer cache completions. (#1564)
+* A failure to parse an ``observe`` mini-language string now raises
+  ``ValueError`` rather than ``LarkError``. (#1507)
+* The ``NotifierNotFound`` exception is now published in
+  ``traits.observation.api``. (#1498)
+* An attempt to access a nonexistent "dunder" attribute (an attribute whose
+  name starts and ends with "__") on a ``CTrait`` instance will now raise
+  ``AttributeError``. Previously, it would return ``None``. (#1477, #1469,
+  #1474)
+
+
+Deprecations
+~~~~~~~~~~~~
+
+* The ``MetaHasTraits.add_listener`` and ``MetaHasTraits.remove_listener``
+  methods are deprecated. If you need the functionality that these methods
+  provide, please discuss your use-case with the ETS maintenance team. (#1550)
+* The ``Symbol`` trait type is deprecated. For resolution of a string
+  representing a package/module/object combination, use ``import_symbol``
+  instead. (#1542)
+* The ``Any`` trait type currently implicitly makes a per-``HasTraits``-instance
```

Comment:
The changes to `Any` and to `Date`, `Datetime` and `Time` should be mentioned more prominently as part of a migration guide. The rest of the deprecations are unlikely to affect most users.